source/GameInterface/Services/Workshops/Patches/WorkshopPropertyPatches.cs

```diff
@@ -114,5 +118,77 @@ private static bool SetInitialCapitalPrefix(Workshop __instance, int value)
         }
+
+        [HarmonyPatch(nameof(Workshop.SetCustomName), MethodType.Normal)]
+        [HarmonyPrefix]
+        private static bool SetCustomNamePrefix(Workshop __instance, TaleWorlds.Localization.TextObject value)
+        {
+            if (CallOriginalPolicy.IsOriginalAllowed()) return true;
+
+            if (ModInformation.IsClient)
+            {
+                Logger.Error("Client tried to set {name}\nCallstack: {callstack}", nameof(Workshop.SetCustomName), Environment.StackTrace);
+                return false;
+            }
+
+            var message = new WorkshopPropertyChanged(PropertyType.CustomName, __instance, value.ToString());
+            MessageBroker.Instance.Publish(__instance, message);
+
+            return ModInformation.IsServer;
+        }
+
+        [HarmonyPatch(nameof(Workshop.Owner), MethodType.Setter)]
+        [HarmonyPrefix]
+        private static bool SetOwnerPrefix(Workshop __instance, Hero value)
+        {
+            if (CallOriginalPolicy.IsOriginalAllowed()) return true;
+
+            if (ModInformation.IsClient)
+            {
+                Logger.Error("Client tried to set {name}\nCallstack: {callstack}", nameof(Workshop.Owner), Environment.StackTrace);
+                return false;
+            }
+
+            var message = new WorkshopPropertyChanged(PropertyType.Owner, __instance, value.StringId);
+            MessageBroker.Instance.Publish(__instance, message);
+
+            return ModInformation.IsServer;
+        }
+
+        [HarmonyPatch(nameof(Workshop.Settlement), MethodType.Setter)]
+        [HarmonyPrefix]
+        private static bool SetSettlementPrefix(Workshop __instance, Settlement value)
+        {
+            if (CallOriginalPolicy.IsOriginalAllowed()) return true;
+
+            if (ModInformation.IsClient)
+            {
+                Logger.Error("Client tried to set {name}\nCallstack: {callstack}", nameof(Workshop.Settlement), Environment.StackTrace);
+                return false;
+            }
+
+            var message = new WorkshopPropertyChanged(PropertyType.Settlement, __instance, value.StringId);
+            MessageBroker.Instance.Publish(__instance, message);
+
+            return ModInformation.IsServer;
+        }
+
+        [HarmonyPatch(nameof(Workshop.Tag), MethodType.Setter)]
+        [HarmonyPrefix]
+        private static bool SetTagPrefix(Workshop __instance, string value)
+        {
+            if (CallOriginalPolicy.IsOriginalAllowed()) return true;
+
+            if (ModInformation.IsClient)
+            {
+                Logger.Error("Client tried to set {name}\nCallstack: {callstack}", nameof(Workshop.Tag), Environment.StackTrace);
+                return false;
+            }
+
+            var message = new WorkshopPropertyChanged(PropertyType.Tag, __instance, value);
+            MessageBroker.Instance.Publish(__instance, message);
+
+            return ModInformation.IsServer;
+        }
```

Comment:
I might have misunderstood something, but don't we need to use the `MessageBroker` here to sync these values? Because i was initially told to just sync all four fields.

If that is the case, would we still not need to sync when the workshop owner changes? There is a setter method for it called `ChangeOwnerOfWorkshop`